reader_concurrency_semaphore.cc

```diff
@@ -1507,6 +1514,12 @@ void reader_concurrency_semaphore::maybe_admit_waiters() noexcept {
         auto& permit = _wait_list.front();
         dequeue_permit(permit);
         try {
+            // Abort the read if its remaining time is less than half of its timeout when arrived to the semaphore
+            if (permit.timeout() - db::timeout_clock::now() <= (permit.timeout() - permit.created())/2) {
+                permit.on_abort();
+                continue;
+            }
```

Comment:
I think the name "abort" can be improved. We don't abort a running query, just a waiting-to-be-admitted query.

Perhaps "unstarted".

Again, why not just throw them away from the queue?

Reply:
The permit is effectively a shared-pointer, and it is not guaranteed that the refcount=1 when the permit is in the queue. It commonly is, but not always: think of re-admitting an evicted read, the permit is referenced by the evictable-reader and the readers above.

Reply:
Can it be simultaneously on the queue and executing? Sounds wrong.

If queued is mutually exclusive with executing, we can just drop the queue entry.

Reply:
Another reason for having a dedicated state, is that currently a timed out permit preserves its state. I managed to hit a `queue overloaded` error when the same permit was accidently dequeued twice and thus `_stats.waiters`, that is of `uint64_t` type, flipped.

Reply:
I think it's okay to have a new state. I imagine someone going through a core dump and being misled by wrong state, better to make it super clear. Even if most of the permits will be destroyed immediately after moving to the new state.